test/EngineTest.cpp

```diff
@@ -10,19 +10,25 @@
 #include "../src/engine/Join.h"
 #include "../src/engine/OptionalJoin.h"
 
+ad_utility::LimitedAllocator<Id>& alloc() {
+  static ad_utility::LimitedAllocator<Id> a{
+      ad_utility::makeAllocationState(4000000000)};
```

Comment:
Why not simply use std::numeric_limits<size_t>::max here and in the other test?

Alternatively, have flag (or special value) in AllocationMemoryLeft, for which the allocation is unlimited.

Reply:
I fully agree and have changed this